src/core/core/address.cpp

```diff
@@ -140,19 +140,25 @@ DSN_API const char*   dsn_address_to_string(dsn_address_t addr)
 {
     char* p = dsn::tls_dsn.scatch_buffer;
     auto sz = sizeof(dsn::tls_dsn.scatch_buffer);
+    struct in_addr net_addr;
+# ifdef _WIN32
+    char* ip_str;
+# else
+    int ip_len;
+# endif
 
     switch (addr.u.v4.type)
     {
     case HOST_TYPE_IPV4:
-        snprintf_p(
-            p, sz,
-            "%u.%u.%u.%u:%hu",
-            ((uint32_t)addr.u.v4.ip >> 24) & 0xff,
-            ((uint32_t)addr.u.v4.ip >> 16) & 0xff,
-            ((uint32_t)addr.u.v4.ip >> 8) & 0xff,
-            ((uint32_t)addr.u.v4.ip) & 0xff,
-            (uint16_t)addr.u.v4.port
-            );
+        net_addr.s_addr = htonl((uint32_t)addr.u.v4.ip);
```

Comment:
addr.u.v4.ip is host address, which may be little endian or big endian depending on the machine. so at least use htonl() to transfer to network address.